include/ring_buffer.h

```diff
@@ -116,6 +149,45 @@ static inline void sys_ring_buf_init(struct ring_buf *buf, u32_t size,
 
 }
 
+/** @brief Determine if head and tail indicates empty ring buffer.
+ *
+ * @note Function for internal use.
+ *
+ * @param head Ring buffer Head.
+ * @param tail Ring buffer Tail.
+ *             Ring buffer
+ * @return 1 if empty, 0 otherwise.
+ */
+static inline int sys_ring_buf_custom_is_empty(u32_t head, u32_t tail)
```

Comment:
I believe the latest convention for internal functions is "z_" prefix (which makes no sense, as that should be *public* prefix, but d'oh, it's Zephyr). @nashif, @andrewboie for consult.

Reply:
> I believe the latest convention for internal functions is "z_" prefix (which makes no sense, as that should be public prefix, but d'oh, it's Zephyr). @nashif, @andrewboie for consult.

@pfalcon this is not funny anymore. This was not made up or decided by me or by @andrewboie, this was a project wide decision. If you have problems with that, you need to bring it forward to the many forums we have, but having these random "unrelated" comments across PRs and Issues about your dislike for certain things is not really helping anyone.

Reply:
@nashif: It seems there's a problem with "accepting criticism". As for "project wide decision", I'm for one wasn't aware of it (like, I didn't see anything in the devel mailing list, or via other "development communication channels"). The only way I'm in loop on it is that in some of @andrewboie's reviews he told me that. That's why I called him and you to confirm if here "sys_ring_buf_custom_is_empty" should instead be "z_sys_ring_buf_is_empty". So, I'm puzzled with your reaction.

Reply:
> It seems there's a problem with "accepting criticism".

I have no problem with criticism, but "d'oh, it's Zephyr" is being sarcastic in a very bad way. As I said, if you have an issue with something, raise it in the right forums and not in random comments across the github project which leads to off-topic discussions and confusion.

There are multiple meetings per week where such issues are being discussed, and everyone is welcome to join them, they are all listed in the wiki. If you want every single decision to be sent to the mailing list, that is a proposal you can make, there is nothing that we are trying to hide, but there is a limit to the channels and methods something can be communicated.

Again, if you think the project is not communicating enough and that we should over-communicate, please raise this in the relevant forums.

Reply:
@nashif: 

> but "d'oh, it's Zephyr" is being sarcastic in a very bad way.

Oh, I'm sorry then, I didn't think your maintainership feelings would be hurt.

> As I said, if you have an issue with something

Let's see. My RFC on namespacing headers/APIs in Zephyr: https://lists.zephyrproject.org/g/devel/topic/16760352#343 , dated 2017-04-01 (not an April 1st joke!). Using "z_" was discussed as some opportunity (in https://lists.zephyrproject.org/g/devel/message/361 specifically). It generated some discussion, but little support. Fast forward a year-something, and turns out that literally the opposite to this RFC was decided: it's now possible to tell if function belongs to Zephyr ... if users are not supposed to use such a function! My comment might be not funny, but this situation is definitely is. And my comment was intended to be just the same - ironic at best, not sarcastic.

> If you want every single decision to be sent to the mailing list

Well, currently I'm just trying to help with putting decisions into practice, which requires help from decision-makers, but so far we discuss procedural matters instead.